File: solutions/java/resistor-color-duo/1/src/main/java/ResistorColorDuo.java
import java.util.List;
import java.util.Arrays;

class ResistorColorDuo {
    final List<String> codes = Arrays.asList("black", "brown", "red", "orange", "yellow", "green", "blue", "violet", "grey", "white");

    int value(String[] colors) {
        return codes.indexOf(colors[0]) * 10 + codes.indexOf(colors[1]);
    }
}

File: solutions/python/resistor-color-duo/1/resistor_color_duo.py
from functools import reduce

COLORS = [
    'black', 'brown', 'red', 'orange', 'yellow',
    'green', 'blue', 'violet', 'grey', 'white',
]
COLORS = {
    color: i
    for i, color in enumerate(COLORS)
}


def value(colors):
    return reduce(
        lambda val, color_val: val * 10 + color_val,
        map(COLORS.get, colors[:2]),
        0
    )